src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -1195,14 +1192,7 @@ SystemDumpMapDCmd::SystemDumpMapDCmd(outputStream* output, bool heap) :
 }
 
 void SystemDumpMapDCmd::execute(DCmdSource source, TRAPS) {
-  stringStream defaultname;
-  const char* name = nullptr;
-  if (_filename.is_set()) {
-    name = _filename.value();
-  } else {
-    defaultname.print("vm_memory_map_%d.txt", os::current_process_id());
-    name = defaultname.base();
-  }
+  const char* name = _filename.value()._name;
```

Comment:
This direct access to the member inside _filename is a bit awkward. I would make the buffer private and give the class some getters and setters, possibly like this:
```
class FileArgument {
  // private stuff
public:
  const char* get() const { 
    // return internal buffer
  }

  // returns true if parsing succeeded, false if not
  bool parse_value(const char* s, size_t len) {
     // call Arguments::copyexpand, target internal buffer, and return its return value
  }
}
```